common/src/chain/transaction/signature/inputsig/mod.rs

```diff
@@ -30,3 +30,12 @@ pub enum InputWitness {
     #[codec(index = 1)]
     Standard(StandardInputSignature),
 }
+
+impl InputWitness {
+    pub fn as_standard_signature(&self) -> Option<&StandardInputSignature> {
+        match self {
+            InputWitness::NoSignature(_) => None,
+            InputWitness::Standard(sig) => Some(sig),
+        }
+    }
+}
```

Comment:
I'm not sure how beneficial this is... I did do this same thing and undid it. The problem here is that NoSignature is also something that needs some level of verification.

In my branch, I added some signature tests, please make sure to cherry pick these commits too.

Reply:
> I'm not sure how beneficial this is... I did do this same thing and undid it.

Apparently it's not even used anymore. Will remove.